packages/react-shopify-app-route-propagator/src/index.ts

```diff
@@ -15,58 +15,38 @@ export interface Props {
 
 export const MODAL_IFRAME_NAME = 'app-modal-iframe';
 
-export default class ReactShopifyAppRoutePropagator extends React.PureComponent<
-  Props,
-  never
-> {
-  private historyInstance: History.History | undefined;
+export default function ReactShopifyAppRoutePropagator({app, location}: Props) {
+  const [history] = useState(History.create(app));
```

Comment:
I think `useRef` is more appropriate here (never changes, old version was an instance property, which is roughly equivalent to a ref)

Reply:
Do we not need to ever respond to a change in `app`? I assume not since the old one doesn't.

Reply:
the old one doesn't, should we? I don't think its typical for app to change after initialization.